nx/lib/nx/container.ex

```diff
@@ -119,7 +119,17 @@ defimpl Nx.Container, for: Any do
         end
       end
 
-    return = struct |> Map.to_list() |> Keyword.merge(full_pattern)
+    return =
+      struct
+      |> Map.to_list()
+      |> Keyword.merge(full_pattern)
+      |> Enum.map(fn {k, v} = x ->
+        if k in keep or k in containers do
```

Comment:
What happens if k in keep but v is a map?

Reply:
I think in those cases it will get normalized properly? I've updated the test to have default values for both a keep and a reset field.